openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java

```diff
@@ -1368,6 +1370,22 @@ private Schema getSchemaDefinition(
         return null;
     }
 
+    private List<String> getEnumValues(EnumElement type) {
+        List<String> enumValues = new ArrayList<>();
+        javax.lang.model.element.Element nativeType = (javax.lang.model.element.Element) type.getNativeType();
+        for (javax.lang.model.element.Element element : nativeType.getEnclosedElements()) {
+            if (element.getKind() == ElementKind.ENUM_CONSTANT) {
+                JsonProperty jsonproperty = element.getAnnotation(JsonProperty.class);
+                if (jsonproperty != null && StringUtils.hasText(jsonproperty.value())) {
+                    enumValues.add(jsonproperty.value());
+                } else {
+                    enumValues.add(element.getSimpleName().toString());
+                }
+            }
+        }
+        return enumValues;
+    }
```

Comment:
This change will only work on Java, we prefer to not have direct dependencies on the `javax.lang.model` api. Can you alter this to use `EnumElement` and the `values()` method is already offers. This possibly will require a change in core.

Reply:
@graemerocher Hmm... if this method cannot be used - then yes, you will have to modify the kernel. Because now it is possible to get only a list of values, without annotations (Ivan described the problem in the issue). In general, need to add a method to `EnumElement` that will return a list of values and their annotations